### ISSUE TITLE
gplearn.fitness.make_fitness code seems have a bug

### ISSUE THREAD

**Issue by LG-1** [trevorstephens/gplearn#97]
if I want to make metric like auc，use the following code to judgment the function return value type may case a bug,
```
    if not isinstance(function(np.array([1, 1]),
                      np.array([2, 2]),
                      np.array([1, 1])), numbers.Number):
        raise ValueError('function must return a numeric.')
```

**Comment by trevorstephens** [comment]
I don't understand at all sorry. What is `function`? Where is this code in your script? Can you provide a complete (short & self contained) example & description of errors to show what issue you are experiencing?

**Comment by LG-1** [comment]
@trevorstephens 
```
from sklearn.metrics import roc_auc_score,roc_curve,auc
from gplearn.fitness import make_fitness
def auc_score(y_true, y_pred, sample_weight=None):
    fpr, tpr, thresholds   = roc_curve(y_true, y_pred, sample_weight)
    return auc(fpr,tpr)
auc_scores = make_fitness(function=auc_score, greater_is_better=True)
```
The above code should reproduce the error

**Comment by trevorstephens** [comment]
I see now. Your metric only creates a warning from scikit-learn's `auc` function so you should still get output. The gplearn fitness function factory and its (incredibly basic) tests are based upon the assumption that you are doing regression as that is what is supported right now. Classification is on the release plan for 0.4.0 and will require me reworking the custom fitness design a bit. In the meantime you could work around this by using the private `_Fitness` object to create your metric manually:

    from gplearn.fitness import _Fitness
    auc_scores = _Fitness(function=auc_score, greater_is_better=True)

Have not tested this on any real problem, but it will bypass the basic tests that are done by `make_fitness` and hopefully give you a pointer in the right direction. Can't guarantee the results will make any sense at all since the estimator is a regressor and you will get potentially unbounded continuous outputs!

In short, classification is not supported yet. Hacking of the metric to make it support is on the user at this point (but hack away, that's always fun) but hang tight and it will be there sooner or later.

**Comment by trevorstephens** [comment]
A quick idea off the top of my head. You might get away with classification by passing `y_pred` through a sigmoid/logit function within the metric definition, and then doing the same to the output of the regressor before consuming it by another step. That's kind of my initial thoughts around how to implement classification in the actual package.

**Comment by LG-1** [comment]
yes, I just Comment this code(that means don't check the function return type anymore):
```
    if not isinstance(function(np.array([1, 1]),
                      np.array([2, 2]),
                      np.array([1, 1])), numbers.Number):
        raise ValueError('function must return a numeric.')
```
and running well🙈🙈
I know not a right way, but quickly😂

**Comment by LG-1** [comment]
@trevorstephens 
it seems nothing helps.
I check to use sigmoid to transform y_pred,but it seems will not change the results not at all.
maybe I think it's because the auc score was computed by sorted data, the sigmoid function is monotonous, will not change the ranks,so will not change the results.
maybe except change the y_pred value use the principle of classification, will not helps.but those maybe need some works.😂😂😂
maybe I'm wrong,Still I am too dishy🙈🙈🙈
